### PR TITLE
Links for international users

### PR DESCRIPTION
**Pull request by Erin007**
Updates to the hamburger for international users, as well as fixing a few style bugs. Combinations can be seen in this [doc](https://docs.google.com/document/d/1hZPjG3OQvLnI8vGIVejUKky44_WRLFj_91lyrFSrU4g/edit).

### CODE REVIEW

**Comment by Erin007** [comment]
cc: @joshlory

File: pegasus/sites.v3/code.org/views/header2.haml [hunk 1]
```diff
@@ -39,6 +39,9 @@
             %a.headerlink{:href=>"/help"}= I18n.t('header_help_us')
             %a.headerlink{:href=>"/about"}= I18n.t('header_about')
             %a.headerlink{:href=>CDO.studio_url("/projects/public")}= I18n.t('header_project_gallery')
+          - else
+            %a.headerlink{:href=>"/courses"}=I18n.t('nav.header.courses')
```
Comment (breville) [inline pegasus/sites.v3/code.org/views/header2.haml]: Do these work?  i think "nav.header." prefix is for dashboard strings, not pegasus.

File: shared/haml/hamburger.haml [hunk 2]
```diff
@@ -4,14 +4,19 @@
     { title: I18n.t("#{loc_prefix}courses"), url: "#{studio_url_base}/courses" },
     { title: I18n.t("#{loc_prefix}project_gallery"), url: "#{studio_url_base}/projects" },
     { title: I18n.t("#{loc_prefix}sections"), url: "#{code_org_url_base}/teacher-dashboard#/sections" },
-    { title: I18n.t("#{loc_prefix}professional_learning"), url: "#{studio_url_base}/my-professional-learning" }
+    { title: I18n.t("#{loc_prefix}professional_learning"), url: "#{studio_url_base}/my-professional-learning" },
```
Comment (breville) [inline shared/haml/hamburger.haml]: unnecessary change?

File: shared/haml/hamburger.haml [hunk 3]
```diff
@@ -40,23 +45,32 @@
...
+  show_help_options = "show-always"
+  show_signed_out_options = "hide-always"
 
   hamburger_class = "none"
 
   if level
     # The header is taken over by level-related UI, so we need the hamburger
     # to show whatever would show up in the header at desktop (and mobile) widths.
     if user_type == "teacher"
+      show_user_options = "show-always"
```
Comment (breville) [inline shared/haml/hamburger.haml]: I wonder if it makes sense to just set `show_user_options = "show-always"` once if there is a level rather than set it in three places here.

File: shared/haml/hamburger.haml [hunk 4]
```diff
@@ -65,9 +79,16 @@
     # appear in the hamburger at mobile widths.
 
     if user_type == "teacher"
+      show_user_options = "show-mobile"
```
Comment (breville) [inline shared/haml/hamburger.haml]: Similarly, we could just have a single `show_user_options = "show-mobile"` instead of three.

File: shared/haml/hamburger.haml [hunk 5]
```diff
@@ -40,23 +45,32 @@
...
+    else
+      show_user_options = "show-always"
+      show_signed_out_options = "show-always"
     end
 
     # Regardless of user type, if they are English, then they also need the pegasus
     # options in the hamburger.
     if language == "en"
       show_pegasus_options = "show-always"
+      show_help_options = "show-always"
```
Comment (breville) [inline shared/haml/hamburger.haml]: I think this is already the default.  Good to make this explicit?

File: shared/haml/hamburger.haml [hunk 6]
```diff
@@ -40,23 +45,32 @@
 
   show_teacher_options = "hide-always"
   show_student_options = "hide-always"
+  show_user_options = "hide-always"
   show_pegasus_options = "hide-always"
+  show_help_options = "show-always"
```
Comment (breville) [inline shared/haml/hamburger.haml]: If we're always going to explicitly set this below, then maybe good to start it as `"hide-always"` so that it's consistent with the others.

File: shared/haml/hamburger.haml (hunk 5 above)
Comment (breville) [reply shared/haml/hamburger.haml]: Alternatively, maybe more consistent to start it as hidden and keep this line here, per comment above.

File: shared/haml/hamburger.haml [hunk 7]
```diff
@@ -82,7 +103,7 @@
   end
 
   # Do we show hamburger on all widths, only mobile, or not at all?
-  if show_teacher_options == "show-always" || show_student_options == "show-always" || show_pegasus_options == "show-always"
+  if show_teacher_options == "show-always" || show_student_options == "show-always" || show_pegasus_options == "show-always" || language != "en"
```
Comment (breville) [inline shared/haml/hamburger.haml]: I would prefer that we key off of another `_show_` variable rather than introducing language into this condition.

Comment (breville) [reply shared/haml/hamburger.haml]: Presumably we can work out what category we're showing for non-en users and use that.

File: shared/haml/hamburger.haml [hunk 8]
```diff
@@ -124,6 +152,8 @@
       - if level || script_level
         %li
           %a{href: CDO.code_org_url('/educate/applab')}=I18n.t("#{loc_prefix}tutorials")
+      -if language == "en"
```
Comment (breville) [inline shared/haml/hamburger.haml]: Again, it would be neat to use a _show_ variable here rather than use implicit knowledge of what appears for an en user.

File: shared/haml/hamburger.haml [hunk 9]
```diff
@@ -92,17 +110,24 @@
 
 #hamburger{class: hamburger_class}
   %ul.hide-responsive-menu
-    .user-links
+    .user-links{class: show_user_options}
```
Comment (breville) [inline shared/haml/hamburger.haml]: `show_user_options` seems redundant since it just wraps some more specific types that we show/hide.

File: shared/haml/hamburger.haml [hunk 10]
```diff
@@ -92,17 +110,24 @@
...
       - elsif user_type == "student"
         - student_entries.each do |entry|
           %li{class: show_student_options}
             %a{href: entry[:url]}= entry[:title]
+        %div{class: "divider"}
+      - else
+        - signed_out_entries.each do |entry|
+          %li{class: show_signed_out_options}
+            %a{href: entry[:url]}= entry[:title]
+        %div{class: "divider"}
```
Comment (breville) [inline shared/haml/hamburger.haml]: An alternative would be to check for type being any of teacher/student/signed_out and then show the divider.

File: shared/haml/hamburger.haml [hunk 11]
```diff
@@ -124,6 +149,8 @@
       - if level || script_level
         %li
           %a{href: CDO.code_org_url('/educate/applab')}=I18n.t("#{loc_prefix}tutorials")
+      -if show_pegasus_options
```
Comment (breville) [inline shared/haml/hamburger.haml]: nit: lead with a space

File: shared/haml/hamburger.haml (hunk 10 above)
Comment (breville) [reply shared/haml/hamburger.haml]: ...or just have a local variable here like `user_items_were_shown` which is set to `true` in any of the above branches, and then we use that to decide whether to show a divider.
